vtex-sites/base.store
# fix: remove from cart and analytics events

## What's the purpose of this pull request?
This PR solves an issue when removing a specific item from cart. For comparison, check the before/after gif below:

<div style="display: flex">
<img src="https://user-images.githubusercontent.com/1753396/155176794-3e568db8-5242-4c88-b9e2-43e8f514c844.gif" alt="before" />
<img src="https://user-images.githubusercontent.com/1753396/155180083-177bbee3-eeac-4a91-b0b2-04a588eb0b4c.gif" alt="after" />
</div>

## How does it work?
We were not retrieving all necessary data from the server on cart validation. This caused, that, for some products, these necessary infos were lacking, that, in turn caused the following null pointer exception on the `useRemoveButton` hook:
<img width="660" alt="image" src="https://user-images.githubusercontent.com/1753396/155177814-61a655c9-524c-4a09-bc5d-4efdb7f4412d.png">

This class of "null pointer exception" errors should be impossible to happen on `base.store` due to the nature of our end-to-end typings powered by GraphQL and TypeScript. Taking a further look, I've figured out that a type cast was the root of the problem. 
The `CartItemWithAnalytics` type, alongside the cast present on `useCart` hook, broke TypeScript typings, making it possible to have objects of incomplete types on the store. To solve the issue, I now removed the type `CartItemWithAnalytics` and added the necessary info to the `validateCart` mutation. 

Also, another problem with this was that we were using the wrong sku id for the product, what caused the wrong product to be added to cart. This PR solves this issue and should be merged before this one
https://github.com/vtex/faststore/pull/1156

## How to test it?
Try to add/remove products from the store and make sure all are removeable, specially this one: `/handcrafted-concrete-soap-36085291/p`


## Checklist
